widgy/contrib/widgy_mezzanine/tests.py

```diff
@@ -264,6 +268,17 @@ def test_legacy_url(self):
                 status_code=301,
             )
 
+    def test_redirects_to_login(self):
+        """
+        Unauthenticated users trying to preview should be redirected to the login page.
+        """
+
+        button = Button.add_root(widgy_site, text='Button text')
+        request = self.factory.get('/')
+        request.user = AnonymousUser()
+        resp = self.preview_view(request, node_pk=button.node.pk)
+        self.assertEqual(resp['Location'], urlresolvers.reverse('login') + '?next=' + request.get_full_path())
+
```

Comment:
maybe this is a bit of an ask, but can you add a test that makes sure an authenticated use gets a 403?

It would be confusing to get redirected to login even when I'm logged in. I know the code already handles this, but it would be nice to make sure it works.